docs/dev/integrate_operator_how_to.md

```diff
@@ -0,0 +1,53 @@
+https://docs.google.com/document/d/1CGZVEyuloZ9oD4NUArW6dnEpi0WFc6BP2tqPSwFZTCY/edit#heading=h.1zgrwxmpgxbr
+
+Upshot of the above document is that the operator should keep its static manifest files in manifests/ folder of the operator repository and manage image references through an image-references file. The order of creation of the files is alphabetical, so the files should be numbered like below to effectively create the service account before the deployment.
+:
+```
+0000_70_my-operator-name_04_service-account.yaml
+0000_70_my-operator-name_05_deployment.yaml
+```
+where, 0000 is the global run-level, 70 is the operator level, and 04/05 is the internal ordering of the resource manifests.
+
+*What to do for my dynamic template files?*
+
+An operator should auto-discover the install config rather than expand the templates through the installer-src integration (see alternative). Simply use the apiserver access to get the install-config as a config map. The config map is stored by the name ‘cluster-config-v1’ in the ‘kube-system’ namespaces.
+Psuedo code:
+```
+kube-client(apiserver-url).Get(Resource: “config-map”, Namespace: “kube-system”, Name: “cluster-config-v1”)
+```
+where, apiserver-url is a cluster supplied ENV var ‘KUBERNETES_SERVICE_HOST’ in the pod.
+Example:
+https://github.com/openshift/machine-config-operator/blob/e932afdec07dc86d5b643590164f86811e205c57/pkg/operator/operator.go#L272
+
+After discovering the InstallConfig, the operator pod can create its own configuration in memory. e.g. the discoverMCOConfig returns the configuration for the operator as it is discovered rather than picking it up from an actual configmap. 
+https://github.com/openshift/machine-config-operator/blob/31c20eefca172d5c1173e7b79b30bad540958dfe/pkg/operator/render.go#L46
+
+#### The alternative (for exceptions only)
+The main directory for manifests in the installer-src is pkg/asset/manifests. Within the installer-src, there are two ways to get the manifests/config files integrated:
+
+ - A new asset for the operator
+Create a new operator asset, and fill up the Dependencies, Name and Generate functions. The Dependencies might contain InstallConfig as an example. In the Generate function, create the config files as asset contents. For the config/manifest’s actual structure, one can choose to vendor the operator’s github pkg and return the entire list of configs and manifests in the Generate function.
+
+ - Template files
+In the pkg/asset/manifests/content/tectonic directory, place the templatized files as golang variables. Then modify pkg/asset/manifests/tectonic.go to expand the template. Expand templateData in template.go for filling up the template variables.
+
+Creating a new asset in the installer-src is only for exceptional cases where CVO cannot take all the required manifests/config files, and auto-discover is not possible or insufficient e.g. the network operator. As another example the machine-config-operator needs TLS config.
```

Comment:
This paragraph should go at the top of its section, so folks who shouldn't be going this route get waved off before working through the details on how this route works.

Reply:
Point taken.